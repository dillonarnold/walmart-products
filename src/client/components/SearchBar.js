import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';

class SearchBar extends Component {
  state = {
    searchString: ''
  };

  handleChange = event => {
    this.setState({
      searchString: event.target.value
    });
  };

  render() {
    return (
      <TextField
        id="search"
        label="Search"
        value={this.state.searchString}
        onChange={this.handleChange}
        margin="normal"
      />
    );
  }
}

export default SearchBar;
