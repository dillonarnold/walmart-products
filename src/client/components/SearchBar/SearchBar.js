import React, { Component } from 'react';
import { func } from 'react-proptypes';
import TextField from '@material-ui/core/TextField';
import { connect } from 'react-redux';
import { searchProducts } from '../../actions/index';

class SearchBar extends Component {
  state = {
    searchString: ''
  };

  static propTypes = {
    searchProducts: func
  };

  static defaultProps = {
    searchProducts: () => {}
  };

  // Used for onEnter on the search text field
  handleTextFieldKeyDown = event => {
    switch (event.key) {
      case 'Enter':
        // Make API call for search query
        this.props.searchProducts(this.state.searchString);
        break;
      default:
        break;
    }
  };

  handleChange = event => {
    this.setState({
      searchString: event.target.value
    });
  };

  render() {
    return (
      <TextField
        id="searchField"
        fullWidth
        label="Search"
        value={this.state.searchString}
        onChange={this.handleChange}
        onKeyDown={this.handleTextFieldKeyDown}
        margin="normal"
      />
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    searchProducts: query => dispatch(searchProducts(query))
  };
};

export default connect(null, mapDispatchToProps)(SearchBar);
