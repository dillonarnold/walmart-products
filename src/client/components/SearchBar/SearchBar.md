Example:

```jsx
<SearchBar searchProducts={(query) => alert(`searchProducts called with ${query}`)} />
```
