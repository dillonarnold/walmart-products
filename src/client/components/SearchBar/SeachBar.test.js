import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBar from './SearchBar';

describe('Product component', () => {

  it('should render properly', () => {
    const component = shallow(
      <SearchBar searchProducts={(test) => {}} />,
    );

    expect(toJson(component)).toMatchSnapshot();

  });

  it('calls search', () => {
    const callback = jest.fn();
    const component = mount(<SearchBar searchProducts={callback} />);
    // Simulate enter key
    component.find('input').simulate('keyDown', {key: 'Enter'});

    expect(callback).toBeCalled();
  });
});
