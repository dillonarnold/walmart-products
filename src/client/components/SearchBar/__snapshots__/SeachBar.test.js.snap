// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render properly 1`] = `
<TextField
  fullWidth={true}
  id="searchField"
  label="Search"
  margin="normal"
  onChange={[Function]}
  onKeyDown={[Function]}
  required={false}
  select={false}
  value=""
/>
`;
